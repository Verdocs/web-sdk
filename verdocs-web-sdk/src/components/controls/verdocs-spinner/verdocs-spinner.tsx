import {Component, h, Host, Prop} from '@stencil/core';

const LIGHT_STYLE = {border: '3px solid rgba(255, 255, 255, 0.3)', borderTopColor: '#fff'};
const DARK_STYLE = {border: '3px solid rgba(0, 0, 0, 0.3)', borderTopColor: '#333'};

/**
 * Display a small loading spinner.
 */
@Component({
  tag: 'verdocs-spinner',
  styleUrl: 'verdocs-spinner.scss',
})
export class VerdocsQuickFilter {
  @Prop()
  size: number = 32;

  @Prop()
  mode: 'light' | 'dark' = 'light';

  render() {
    return (
      <Host
        style={{
          width: `${this.size}px`, //
          height: `${this.size}px`,
          flex: `0 0 ${this.size}px`,
          ...(this.mode === 'light' ? LIGHT_STYLE : DARK_STYLE),
        }}
      />
    );
  }
}
