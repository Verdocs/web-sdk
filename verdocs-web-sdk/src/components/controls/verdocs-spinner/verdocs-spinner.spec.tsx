import { newSpecPage } from '@stencil/core/testing';
import { VerdocsQuickFilter } from './verdocs-spinner';

describe('verdocs-spinner', () => {
  it('renders with default props', async () => {
    const page = await newSpecPage({
      components: [VerdocsQuickFilter],
      html: `<verdocs-spinner></verdocs-spinner>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders with custom size and mode', async () => {
    const page = await newSpecPage({
      components: [VerdocsQuickFilter],
      html: `<verdocs-spinner size="48" mode="dark"></verdocs-spinner>`,
    });
    expect(page.root).toMatchSnapshot();
    expect(page.rootInstance.size).toBe(48);
    expect(page.rootInstance.mode).toBe('dark');
  });
});
