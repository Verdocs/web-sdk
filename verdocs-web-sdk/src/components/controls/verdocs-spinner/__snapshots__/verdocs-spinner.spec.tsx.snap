// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verdocs-spinner renders with custom size and mode 1`] = `<verdocs-spinner mode="dark" size="48" style="width: 48px; height: 48px; flex: 0 0 48px; border: 3px solid rgba(0, 0, 0, 0.3); border-top-color: #333;"></verdocs-spinner>`;

exports[`verdocs-spinner renders with default props 1`] = `<verdocs-spinner style="width: 32px; height: 32px; flex: 0 0 32px; border: 3px solid rgba(255, 255, 255, 0.3); border-top-color: #fff;"></verdocs-spinner>`;
